bindings/rust/bench/scripts/install-aws-lc.sh

```diff
@@ -3,34 +3,44 @@
 # Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 # SPDX-License-Identifier: Apache-2.0
 
+# Usage: ./install-aws-lc.sh
 # Sets up bench crate to use aws-lc for either s2n-tls or rustls if desired
-# To run with aws-lc, use any cargo command with:
+
+# To run benches with aws-lc, use any cargo command with:
 # --config aws-lc-config/s2n.toml
 # --config aws-lc-config/rustls.toml
 # or both
 
+# How Rustls with aws-lc-rs works:
+# Clones aws-lc-rs, changes its name to ring with a compatible version number, and
+# patches it into Rustls
+
+# How s2n-tls with aws-lc works:
+# Builds s2n-tls static lib with AWS-LC interned, required to avoid symbol collisions with OpenSSL
+# Checks for libs2n.a at target/s2n-tls-build/lib/libs2n.a
+# Checks for libcrypto.a at target/aws-lc/install/lib/libcrypto.a
+
 set -e 
 
 # go to bench directory
-pushd "$(dirname "$0")" > /dev/null
+pushd "$(dirname "$0")"/.. > /dev/null
 bench_dir="$(pwd)"
 
 
 
 # ----- rustls -----
 
-# clone rustls to bench/target/rustls and checkout compatible version
-rm -rf target/rustls
-git clone https://github.com/rustls/rustls target/rustls
-cd target/rustls
-git checkout 'v/0.21.5'
-
-# go to dir with rustls crate
-cd rustls
-rustls_dir="$(pwd)"
+# clone aws-lc-rs to target
+rm -rf target/aws-lc-rs
+git clone https://github.com/aws/aws-lc-rs target/aws-lc-rs
+cd target/aws-lc-rs/aws-lc-rs
+git submodule init
+git submodule update
 
-# change rustls to use aws-lc-rs
-sed -i 's|ring = .*|ring = { package = "aws-lc-rs" }|' Cargo.toml
+# change aws-lc-rs to look like API compatible ring (name and version)
+# changes first occurrence of 'name = .*' and 'version = .*' to be 
+# 'name = "ring"' and 'version = "0.16.20"'
```

Comment:
What's the significance of 0.16.20? Will future aws-lc-rs versions continue to be API-compatible with this version of rustls? I wonder if the aws-lc-rs version should be pinned to make sure an update won't break this?

Reply:
aws-lc-rs "strives" to be API compatible with ring. Even if aws-lc-rs becomes incompatible with ring one day, it'll probably be accidental and the benches failing will just be another sign of it.

The ring version being always 0.16.20 I'm not as comfortable with since Rustls makes no promises to have a dependency to the same version of ring, so I added some small logic to find the version the Rustls wants and hacks aws-lc-rs's version to that one instead.